src/Type/ObjectType.php

```diff
@@ -156,7 +156,7 @@ public function hasProperty(string $propertyName): TrinaryLogic
 		}
 
 		$classHasProperty = RecursionGuard::run($this, static fn (): bool => $classReflection->hasProperty($propertyName));
-		if ($classHasProperty === true || $classHasProperty instanceof ErrorType) {
+		if ($classHasProperty !== false) {
```

Comment:
are we sure this does not change existing logic?

Reply:
Yeah, definitely weird